### DISCUSSION TITLE
Integrate Gaphor with Matlab

### DISCUSSION THREAD

**Discussion by paulcuda** [gaphor/gaphor#3374] [Ideas]
Matlab makes it impossible to learn their equivalent application "System Composer".  And if you possess only a Student or Home license, they deny LEARNING composer.  It's not a good idea to compete with all other MatLab products, but synergy would be a plus for all considered....especially the MBSE users (potential or current).
A monumental task I know....

**Comment by danyeaw** [comment]
Hi @paulcuda, thanks for the suggestion. To summarize it to make sure I understand, are you looking for an integration between Gaphor and Matlab / Simulink? I know that Mathworks pushes Matlab pretty hard to universities and students, but I am not sure that we want to spend our free time creating integrations to proprietary tools unless a company wants to fund it. What would that integration look like?

**Comment by paulcuda** [comment]
I fully understand this pragmatic approach.  Just wanted to throw it out for consideration.  I’m known for crazy ideas…..some of which have been quite fruitful.

**Comment by TKatzwinkel** [comment]
Hi there,

how about integrating with Scilab / Xcos then? This is somewhat the open source equivalent of MatLab / Simulink.

The idea of integration ist described here in theory (although not with Gaphor in mind): https://scilab.gitlab.io/legacy_wiki/Xcos(2f)SysML

In example, a SysML Block with Input and Output could be passed from Gaphor towards Scilab/Xcos, so all definitions and attributes within the simulation stick to the system engineer's model. So we have consistency within all sub-models, which is Part of the whole MBSE idea.

This could be useful e.g. to check if requirements are satisfied by simulation data and will avoid human mistakes in the copied model data (e.g. flow names, part properties).

Also, existing simulation models could be referred to within the gaphor diagrams by importing them as a block with suiting flow names.

**Comment by paulcuda** [comment]
Thank you….I will examine your recommendations.

Its good to know that someone is thing about the same thing.

Paul

**Comment by paulcuda** [comment]
Send me more info if you discover anything.  MatLab is the gorilla in the tent….Like some competition.
